tinygrad/tensor.py

```diff
@@ -366,9 +366,11 @@ def unsqueeze(self, dim):
     return self.reshape(self.shape[:dim] + (1,) + self.shape[dim:])
 
   # (padding_left, padding_right, padding_top, padding_bottom)
-  def pad2d(self, padding:Union[List[int], Tuple[int, ...]]):
-    slc = [(-p0, s+p1) for p0,p1,s in zip(padding[::2], padding[1::2], self.shape[::-1])][::-1]
-    return self.slice([(0,s) for s in self.shape[:-(len(padding)//2)]] + slc)
+  def pad2d(self, padding:Union[List[int], Tuple[int, ...]], value:int=0):
+    shape = [(0,s) for s in self.shape[:-(len(padding)//2)]]
+    slc = shape + [(-p0, s+p1) for p0,p1,s in zip(padding[::2], padding[1::2], self.shape[::-1])][::-1]
+    if value == 0: return self.slice(slc)
+    return self.slice(slc) + (Tensor.ones(self.shape).slice(slc) > 0).where(0, value) #ones for binary like map
```

Comment:
`(Tensor.ones(self.shape).slice(slc) > 0)` isn't this always going to be true?

Reply:
Nope, .slice adds 0 padding so there will be 1s and 0s, main reason where flips 1s with 0s.
Any suggestion for flipping without zero

Reply:
`(1 - Tensor.ones(self.shape).slice(slc)) * value`

Reply:
Makes too much sense, my bad